src/RedisOutputCachingMiddleware/RedisOutputCachingMiddleware.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.Web.Redis;
+using StackExchange.Redis;
+
+namespace RedisOutputCachingMiddleware
+{
+    public class OutputCachingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IDatabase _cache;
+        // optional expiration time, default to 1 day if not defined 
+        private int _ttl = (int)TimeSpan.FromDays(1).TotalSeconds;
+        
+        public OutputCachingMiddleware(RequestDelegate next, string redisConnectionString, [Optional] int ttl)
+        {
+            _next = next;
+            try
+            {
+                _cache = ConnectionMultiplexer.Connect(redisConnectionString).GetDatabase();
+            }
+            catch (Exception ex)
+            {
+                LogUtility.LogError($"Cannot connect to Redis: {ex.Message}");
+            }
+            
+            _ttl = ttl;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            // use the url, header, and request body as a key 
+            string pathAndQuery = context.Request.GetEncodedPathAndQuery();
+            string headers = context.Request.Headers.ToString();
+            string body = context.Request.Body.ToString();
+            RedisKey key = $"{pathAndQuery}{headers}{body}";
+            RedisValue value = await GetCacheAsync(key);
+
+            if (!value.IsNullOrEmpty)
+            {
+                await context.Response.WriteAsync(value);
+                return;
+            }
+
+            HttpResponse response = context.Response;
+            Stream originalStream = response.Body;
+
+            try
+            {
+                using (var ms = new MemoryStream())
+                {
+                    // record the output stream
+                    response.Body = ms;
+                    // get the new page
```

Comment:
Are we really getting the next page? I was assuming that calling `_next` invokes the next middleware (or action)